### ISSUE TITLE
Add a Resource to handle project files/directories references

### ISSUE THREAD

**Issue by simo498** [godotengine/godot-proposals#9552]
References my open discussion https://github.com/godotengine/godot-proposals/discussions/9545#discussion-6524297

### Describe the project you are working on

An RPG game, where any of the adjacent rooms are loaded at runtime

### Describe the problem or limitation you are having in your project

Each of the rooms needs to keep the reference to the adjacent rooms as a string path, but the only way assign file paths to a property is by passing a string representing it. You can use an attribute hint to make it easier:

```c#
[Export(PropertyHint.File)] // or [Export(PropertyHint.Dir)]
public string pathToResource; //or pathToPackedScene, or pathToDir
```
but it's still not possible to keep track of files/directories which get renamed or moved.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If you use a resource directly, the engine is able to keep track of it even when renamed or moved inside the editor, i'd like to implement something similar but limiting this feature to keep track of a (project) resource path (or directory), to give the developer a way to manually handle each resource (when load/unload them, and how to do it, for example using other threads to load some scenes and resources) 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- This functionality should be added as a new resource type (something like `Path`), which only takes a string as its member fields.

- The user should be able to pass a file/directory into this resource from the editor, and the latter should be tracked by the editor and updated whether it gets moved or renamed from inside the editor.

- The user should be able to access the file path as a string with a method

- It should be possible to view the preview of the resource (such as the image thumbnail or similar)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can't be worked around. The only way to achieve the same result is by manually changing each of the references to a certain file/directory when moved/renamed (which is not convenient at all, in particular when you are moving a large amount of file into a new location, and might lead to some of the references to be null if you forget to manually update them)

### Is there a reason why this should be core and not an add-on in the asset library?

It should improve the editor usability, as it's a missing core funcionality

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot-proposals/issues/9216

**Comment by huwpascoe** [comment]
I've created a plugin that should satisfy this issue.
https://github.com/godotengine/godot/pull/91815#issuecomment-2323095285

**Comment by huwpascoe** [comment]
@simo498 make sure the plugin is enabled in project settings

**Comment by huwpascoe** [comment]
@simo498 Yes, it only changes the editor behavior so it can be removed at export time.

**Comment by huwpascoe** [comment]
Oh! @simo498 check the latest version of the plugin, I think i fixed why it wasn't working for you
